src/methods/large_shuffle.jl

```diff
@@ -19,74 +33,99 @@ Hence, these surrogates can be used to test any null hypothesis aimed at
 comparing short-range dynamical properties versus long-range dynamical
 properties of the signal.
 """
-struct BlockShuffle <: Surrogate
-    n::Int
-end
-
-Base.show(io::IO, bs::BlockShuffle) = show(io, "BlockShuffle(n=$(bs.n))")
+struct BlockShuffle{I <: Integer, B <: Bool} <: Surrogate
+    n::I
+    shift::B
 
-# Split time series in two by default.
-BlockShuffle() = BlockShuffle(2)
+    function BlockShuffle(n::I; shift::B = false) where {I <: Integer, B <: Bool}
+        return new{I, B}(n, shift)
+    end
 
-function get_uniform_blocklengths(L::Int, n::Int)
-    # Compute block lengths
-    N = floor(Int, L/n)
-    R = L % n
-    blocklengths = [N for i = 1:n]
-    for i = 1:R
-        blocklengths[i] += 1
+    # Split time series into 10 pieces by default. Shifting disabled by default.
+    function BlockShuffle()
+        return new{Int, Bool}(10, false)
     end
-    return blocklengths
 end
 
 function surrogenerator(x::AbstractVector, bs::BlockShuffle, rng = Random.default_rng())
-    L = length(x)
-    bs.n < L || error("The number of blocks exceeds number of available points")
-    Ls = get_uniform_blocklengths(L, bs.n)
-    cs = cumsum(Ls)
-    # will hold a rotation version of x
-    xrot = similar(x)
-    T = eltype(xrot)
-    init = NamedTuple{(:L, :Ls, :cs, :xrot),Tuple{Int, Vector{Int}, Vector{Int}, Vector{T}}}((L, Ls, cs, xrot))
-    return SurrogateGenerator(bs, x, init, rng)
-end
+    bs.n < length(x) || error("The number of blocks exceeds number of available points")
 
-function (bs::SurrogateGenerator{<:BlockShuffle})()
-    # TODO: A circular custom array implementation would be much more elegant here
-    L = bs.init.L
-    Ls = bs.init.Ls
-    cs = bs.init.cs
-    xrot = bs.init.xrot
-    n = bs.method.n
-    x = bs.x
-
-    # Just create a temporarily randomly shifted array, so we don't need to mess
-    # with indexing twice.
-    circshift!(xrot, x, rand(bs.rng, 1:L))
-
-    # Block always must be shuffled (so ordered samples are not permitted)
-    draw_order = zeros(Int, n)
-    while any(draw_order .== 0) || all(draw_order .== 1:n)
-       StatsBase.sample!(bs.rng, 1:n, draw_order, replace = false)
-    end
+    # The lengths of the blocks (one block will have a differing length if length of 
+    # time series in not a multiple of the number of blocks, so )
+    blocklengths = get_uniform_blocklengths(length(x), bs.n)
+
+    # The start index of each block.
+    startinds = startinds = [1; cumsum(blocklengths) .+ 1]
```

Comment:
This is more like a remark not a comment to be addressed. I think it would be more intuitive if you wrote this as `vcat(1, cumsum...)`. The reason being that brackets and `;` are used in matrix construction `[1 2; 3 4]`. But here you are actually making a vector. I first thought "he's making an array, but why?" but took me a bit of time to realize its a vector. With `vcat` there is no ambiguity really.

(p.s.: an alternative to vcat would be `pushfirst!(cumsum(blocklengths) .+ 1, 1)`

Reply:
Sure, we can go for `vcat(1, cumsum...)`. I agree that it makes it clearer that this is a vector in this context.

Reply:
to be clear: `vcat(1, cumsum(blocklengths) .+ 1)`.